CONTRIBUTING.md

```diff
@@ -0,0 +1,140 @@
+1. Then, [clone the repository](https://help.github.com/articles/cloning-a-repository/) to your local machine so you can begin making changes.
+1. Add this repository as an [upstream remote](https://help.github.com/en/articles/configuring-a-remote-for-a-fork) on your local git repository so that you are able to fetch the latest commits.
+1. On your local machine make sure you have the latest version of the `develop` branch:
+
+    ```shell
+    git checkout develop
+    git pull upstream develop
+    ``` -->
+
+### Install development dependencies
+
+Please follow the [README.md](README.md) to install the package for development purposes.
+
+### Developing at Sage Bio
+
+The agora-data-tools project follows the standard [trunk based development](https://trunkbaseddevelopment.com/) development strategy.
+> To ensure the most fluid development, do not push to your `dev`
+
+1. Please ask for write permissions to contribute directly to this repository.
+1. Make sure you are always creating feature branches from the `dev` branch.
+
+    ```shell
+    git checkout dev
+    git pull
+    ```
+
+1. Create a feature branch from the `dev` branch. Use the Id of the JIRA issue that you are addressing and name the branch after the issue with some more detail like `{user}/{JIRA}-123/add-some-new-feature`.
+
+    ```shell
+    git checkout dev
+    git checkout -b tyu/JIRA-123/some-new-feature
+    ```
+
+1. At this point, you have only created the branch locally, you need to push this to your fork on GitHub.
+
+    ```shell
+    git push --set-upstream origin tyu/JIRA-123/some-new-feature
+    ```
+
+    You should now be able to see the branch on GitHub. Make commits as you deem necessary. It helps to provide useful commit messages - a commit message saying 'Update' is a lot less helpful than saying 'Remove X parameter because it was unused'.
+
+    ```shell
+    git commit changed_file.txt -m "Remove X parameter because it was unused"
+    git push
+    ```
+
+1. (Make sure you have follow instructions in "Install development dependencies") Once you have made your additions or changes, make sure you write tests and run the test suite.  More information on testing below.
+
+    ```shell
+    pytest -vs tests/
+    ```
+
+1. (Make sure you have follow instructions in "Install development dependencies") Make sure to run the auto python code formatter, black.
+
+    ```shell
+    black ./
+    ```
+
+1. Once you have completed all the steps above, create a pull request from the feature branch to the `dev` branch of the Sage-Bionetworks/agora-data-tools repo.
+
+> *A code maintainer must review and accept your pull request.* A code review ideally happens with both the contributor and the reviewer present, but is not strictly required for contributing. This can be performed remotely (e.g., Zoom, Hangout, or other video or phone conference).  Most code reviews are done asyncronously.
```

Comment:
I'm a little confused by the first half of this because we actually _don't_ do reviews via Zoom etc at all. Or at least, Jess and I don't?